pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -78,6 +82,12 @@ rec {
 
     , ... } @ attrs:
 
+    # Check that the name is consistent with pname and version:
+    assert lib.assertMsg
+      (lib.lists.all (name: builtins.hasAttr name attrs) ["name" "pname" "version"]
+        -> lib.strings.hasSuffix "${attrs.pname}-${attrs.version}" attrs.name)
+      "mkDerivation: `name` must be consistent with `pname-version`";
```

Comment:
I’d always include what the actual values are, otherwise they get lost in the stack traces. So in this case:

```
"mkDerivation: `name` ("${attrs.name}") must be consistent with `pname-version` ("${attrs.pname}-${attrs.version}").";
```
(untested)

Though I still think the check should disallow having both `name` and `pname`/`version`.

Reply:
Solved in 1f7fc09